addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -767,339 +1183,339 @@ def test_assert_created_backorders_computed_correctly(self):
-#         self.assertEqual(pick.state, "done")
-
-#         # Check the total quantity in the picking, relative to the UoM of the move
-#         # not the products.
-#         # So this is the sum of the product_uom_qty of the moves.
-#         self.assertEqual(pick.u_total_quantity, 12)
-#         self.assertEqual(pick.u_quantity_done, 12)
-
-
-# class TestStockPickingUnlinkingEmpties(common.BaseUDES):
-#     """
-#     Tests for unlinking empty pickings functionality.
-
-#     This class tests StockPicking.unlink_empty() directly, and indirectly
-#     StockPicking.get_empty_pickings().
-#     """
-
-#     def test_unlinks_tagged_empty_pickings_in_recordset(self):
-#         """Empty pickings with u_is_empty = True in a recordset are unlinked."""
-#         pick = self.create_picking(self.picking_type_pick)
-#         pick.u_is_empty = True
-
-#         result = pick.unlink_empty()
-
-#         self.assertFalse(result)
-
-#     def test_does_not_unlink_untagged_empty_picking(self):
-#         """An empty picking with u_is_empty = False will not be deleted."""
-#         pick = self.create_picking(self.picking_type_pick)
-#         pick.u_is_empty = False
-
-#         result = pick.unlink_empty()
-
-#         self.assertEqual(pick, result)
-
-#     def test_raises_exception_on_unlink_non_empty_pickings(self):
-#         """Non-empty pickings in a recordset are not unlinked."""
-#         pick = self.create_picking(
-#             self.picking_type_pick, products_info=[{"product": self.apple, "uom_qty": 1}]
-#         )
-#         pick.u_is_empty = True
-
-#         with self.assertRaisesRegex(
-#             ValidationError, r"Trying to unlink non empty pickings: \[\d+\]"
-#         ):
-#             pick.unlink_empty()
+    def test_assert_picking_quantities_computed_correctly(self):
+        """Assert that qty todo/done and package discrepancies fields are computed correctly"""
+        apple_qty = 10
+        apple_qty_per_line = apple_qty / 2
+
+        # Create quants for apples in two separate packages
+        pack1 = self.create_package()
+        pack2 = self.create_package()
+        self.create_quant(
+            self.apple.id, self.test_stock_location_01.id, apple_qty_per_line, package_id=pack1.id
+        )
+        self.create_quant(
+            self.apple.id, self.test_stock_location_01.id, apple_qty_per_line, package_id=pack2.id
+        )
+
+        # Create picking with demand for all apples in stock, split over 2 lines
+        pick = self.create_picking(
+            self.picking_type_pick,
+            products_info=[
+                {"product": self.apple, "uom_qty": apple_qty_per_line},
+                {"product": self.apple, "uom_qty": apple_qty_per_line},
+            ],
+            location_dest_id=self.test_received_location_01.id,
+            location_id=self.test_stock_location_01.id,
+            assign=True,
+        )
+
+        self.assertEqual(pick.u_quantity_done, 0, "Quantity done should be zero")
```

Comment:
I think it's more useful here and below to use the default assert message: I'd rather see the incorrect value than a message that tells me it's wrong without displaying the value.

Reply:
Yep agree,  think i may have commented this out forcing it to appear in the CR. My own silly mistake lol